packages/vc-status-list-issuer-rest-api/src/api-functions.ts

```diff
@@ -96,6 +102,84 @@ export function getStatusListCredentialIndexStatusEndpoint(router: Router, conte
     console.log(`Get statusList credential index status endpoint is disabled`)
     return
   }
+
+  const path = opts?.path ?? '/status-lists/:statusListId/entry/:entryId'
+  router.get(path, checkAuth(opts?.endpoint), async (request: Request, response: Response) => {
+    try {
+      const statusListIdType = (request.query.statusListIdType as StatusListIdType) ?? StatusListIdType.StatusListId
+      const entryIdType = (request.query.entryIdType as EntryIdType) ?? EntryIdType.StatusListIndex
+
+      let statusListIndex: number | undefined
+      let entityCorrelationId: string | undefined
+      let statusListId: string | undefined
+      let statusListCorrelationId: string | undefined
+
+      if (entryIdType === EntryIdType.StatusListIndex) {
+        try {
+          statusListIndex = Number.parseInt(request.params.entryId)
+          if (!statusListIndex || statusListIndex < 0) {
+            return sendErrorResponse(response, 400, `Please provide a proper statusListIndex`)
+          }
+        } catch (error) {
+          return sendErrorResponse(response, 400, `Please provide a proper statusListIndex`)
+        }
+      } else {
+        entityCorrelationId = request.params.entryId
+      }
+
+      if (statusListIdType === StatusListIdType.StatusListId) {
+        statusListId = request.params.statusListId
+      } else {
+        statusListCorrelationId = request.params.statusListId
+      }
+
+      const driver = await getDriver({
+        ...(statusListCorrelationId ? { correlationId: statusListCorrelationId } : { id: statusListId }),
+        dbName: opts.dbName,
+      })
+
+      const details = await driver.getStatusList()
+      if (statusListIndex && statusListIndex > details.length) {
+        return sendErrorResponse(response, 400, `Please provide a proper statusListIndex`)
+      }
+
+      let entry = await driver.getStatusListEntryByIndex({
+        statusListId: details.id,
+        ...(entityCorrelationId ? { correlationId: entityCorrelationId } : { statusListIndex }),
+        errorOnNotFound: false,
+      })
+
+      const type = details.type === StatusListType.StatusList2021 ? 'StatusList2021Entry' : details.type
+      const resultStatusIndex = entry?.statusListIndex ?? statusListIndex ?? 0
+      const status = await checkStatusIndexFromStatusListCredential({
+        statusListCredential: details.statusListCredential,
+        ...(details.type === StatusListType.StatusList2021 ? { statusPurpose: details.statusList2021?.statusPurpose } : {}),
+        type,
+        id: details.id,
+        statusListIndex: resultStatusIndex,
+      })
+
+      if (!entry) {
+        entry = {
+          statusList: details.id,
+          value: '0',
+          statusListIndex: resultStatusIndex,
+        }
+      }
+
+      response.statusCode = 200
+      return response.send({ ...entry, status })
```

Comment:
response.json. To make sure we always have the correct headers

Reply:
fixed (though response.send does detect it's JSON atm.)

Reply:
No, we have seen examples where this returned html all of a sudden